libobs/util/platform-nix.c

```diff
@@ -96,10 +99,56 @@ void os_dlclose(void *module)
 		dlclose(module);
 }
 
+int module_has_qt5_check(const char *path)
+{
+	void *mod = os_dlopen(path);
+	if (mod == NULL) {
+		return 1;
+	}
+
+	struct link_map *list = NULL;
+	if (dlinfo(mod, RTLD_DI_LINKMAP, &list) == 0) {
+		for (struct link_map *ptr = list; ptr; ptr = ptr->l_next) {
+			char *slash = strrchr(ptr->l_name, '/');
+			if (!slash)
+				continue;
+			char *base = slash + 1;
+			if (strncmp(base, "libQt5", 6) == 0) {
+				return 0;
+			}
+		}
+	}
+
+	return 1;
+}
+
+void fork_and_qt5_check(const char *path, bool *can_load)
+{
+	pid_t pid = fork();
+	if (pid == 0) {
+		_exit(module_has_qt5_check(path));
+	}
+	if (pid < 0) {
+		return;
+	}
+	int status;
+	if (waitpid(pid, &status, 0) < 0) {
+		return;
+	}
+	if (WIFEXITED(status) && WEXITSTATUS(status) == 0) {
+		*can_load = false;
+	}
+}
+
 void get_plugin_info(const char *path, bool *is_obs_plugin, bool *can_load)
 {
 	*is_obs_plugin = true;
+#if OBS_QT_VERSION == 6
 	*can_load = true;
+	fork_and_qt5_check(path, can_load);
+#else
+	*can_load = true;
```

Comment:
No need to have `*can_load = true;` two times.

```suggestion
	*is_obs_plugin = true;
	*can_load = true;
#if OBS_QT_VERSION == 6
	fork_and_qt5_check(path, can_load);
```